File: core/server/project/commonBase/src/main/java/com/home/commonBase/scene/base/RoleLogicBase.java
package com.home.commonBase.scene.base;

import com.home.commonBase.data.scene.role.SceneRoleData;

/** 场景玩家逻辑基类 */
public class RoleLogicBase extends SceneObjectLogicBase
{
	/** 角色 */
	protected Role _role;
	/** 数据 */
	protected SceneRoleData _data;
	
	@Override
	public void setObject(SceneObject obj)
	{
		super.setObject(obj);
		
		_role=(Role)obj;
	}
	
	@Override
	public void init()
	{
		super.init();
		_data=_role.getData();
	}
	
	@Override
	public void dispose()
	{
		super.dispose();
		_data=null;
	}
}

File: core/client/game/src/commonGame/scene/base/RoleLogicBase.cs
using System;
using ShineEngine;

/// <summary>
/// 场景角色逻辑
/// </summary>
public class RoleLogicBase:SceneObjectLogicBase
{
	/** 角色 */
	protected Role _role;
	/** 数据 */
	protected SceneRoleData _data;

	public override void setObject(SceneObject obj)
	{
		base.setObject(obj);

		_role=(Role)obj;
	}

	public override void init()
	{
		base.init();
		_data=_role.getData();
	}

	public override void dispose()
	{
		base.dispose();
		_data=null;
	}
}